pmd-core/src/main/java/net/sourceforge/pmd/reporting/FileAnalysisListener.java

```diff
@@ -90,10 +90,10 @@ static FileAnalysisListener tee(Collection<? extends FileAnalysisListener> liste
         List<FileAnalysisListener> list = new ArrayList<>(listeners);
         list.removeIf(it -> it == NoopFileListener.INSTANCE);
 
-        if (listeners.isEmpty()) {
+        if (list.isEmpty()) {
             return noop();
-        } else if (listeners.size() == 1) {
-            return listeners.iterator().next();
+        } else if (list.size() == 1) {
+            return list.iterator().next();
         }
```

Comment:
I just figured this was not actually looking into the filtered list

Reply:
Nice catch, thanks!